app/src/main/java/chat/rocket/android/authentication/login/presentation/LoginPresenter.kt

```diff
@@ -56,4 +61,15 @@ class LoginPresenter @Inject constructor(private val view: LoginView,
     fun signup() {
         navigator.toSignUp()
     }
+
+    private suspend fun registerPushToken() {
+        // TODO: put it on constructor
+        val localRepository: LocalRepository = SharedPreferencesRepository(navigator.activity)
```

Comment:
You should inject the Repository into the Presenter, and the (application)Context can be injected into the repository.

The presenter should never directly know or access any context, Activity or Application

Reply:
Actually, we should make RocketChatApplication a singleton.

Reply:
it is already, just as a `Context`
```
    @Provides
    @Singleton
    fun provideContext(application: Application): Context {
        return application
    }
```

Reply:
But either way, the presenter should never directly see or touch the Context...